### PR TITLE
Fix access of 0D view data

### PR DESCRIPTION
**Pull request by MattRolchigo**
Access 0D view data using "Data(0)" rather than "Data()", to avoid Kokkos error

### CODE REVIEW

File: src/CAinitialize.cpp [hunk 1]
```diff
@@ -1534,20 +1534,29 @@ void ActiveCellInit(int id, int MyXSlices, int MyYSlices, int nz, ViewI_H CellTy
...
-                            int NeighborD3D1ConvPosition =
-                                MyNeighborZ * MyXSlices * MyYSlices + MyNeighborX * MyYSlices + MyNeighborY;
-                            if ((MyNeighborX >= 0) && (MyNeighborX < MyXSlices) && (MyNeighborY >= 0) &&
-                                (MyNeighborY < MyYSlices) && (MyNeighborZ >= 0) && (MyNeighborZ < nz)) {
-                                if (CellType(NeighborD3D1ConvPosition) == Solid) {
-                                    // This cell is at the interface - becomes active type
-                                    CellType(CAGridLocation) = Active;
-                                    SubstrateActCells_ThisRank++;
-                                    return;
+                        if (k == 0) {
```
Comment (streeve) [inline src/CAinitialize.cpp:1537]: This seems unrelated

Comment (MattRolchigo) [reply src/CAinitialize.cpp:1537]: That is unrelated, that change is the subject of a different PR. I must have messed up when rebasing

**Review by streeve — APPROVED** [review]
